app/models/ability.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    case
+    when user.kid?
+      can :show,   User, id: user.id
+      can :manage, Gift, added_by: user
+      can :manage, Image, gift: { added_by: user }
+    when user.elf?
+      can :manage, User, role: 'kid'
+      can :read,   Gift
+      can %i[update create destroy], Gift, added_by: user
+      can :manage, Image, gift: { added_by: user }
+      can %i[show create destroy], Assessment, author: user
+      can :manage, Review, reviewer: user
+    when user.santa?
+      can :index,  User, role: %w[kid elf]
+      can :read,   User, role: 'kid'
+      can :read,   Gift
+      can :read,   Assessment
+      can %i[read destroy], Review
+    when user.admin?
```

Comment:
Shouldn't santa role be equal to admin? Maybe you should allow everything for santa and remove admin

Reply:
I assumed that the only functions Santa should be able to have are review moderation and final gifts selection for kids. He can't assess or review kids and add gifts to elves or to himself. I added an admin role solely for the purpose of testing. That's why i was unsure of my implementation of tests.
If it's okay for Santa to assess the kids himself and have his own wish list i'll replace an admin role with santa.

Reply:
* Kids should have only their functionality with creating gift wishes
* Elves should be able to review them and create a gift proposals
* Santa should be able to do all the things that elves can + his own functionality
* Santa and elves should not be able to kids things

You can leave the admin role for convenience if you need it